spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -24,32 +34,29 @@
     end
 
     context "when sending a welcome_thread notification" do
-      before do
-        welcome_thread_article = create(:article, title: "Welcome Thread - v0", published: true, tags: "welcome", user: mascot_account)
-        create(:comment, commentable: welcome_thread_article, commentable_type: "Article", user: user)
-      end
-
       it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
-        expect(receiving_user.notifications.count).to eq(0)
-        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+        expect do
+          sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+        end.to change(user.notifications, :count).by(1)
 
-        expect(receiving_user.notifications.count).to eq(1)
-        expect(receiving_user.notifications.first.notifiable).to eq(welcome_broadcast)
+        expect(user.notifications.first.notifiable).to eq(welcome_broadcast)
       end
 
       it "does not send a notification to a user who has commented in a welcome thread", :aggregate_failures do
-        expect(user.notifications.count).to eq(0)
-        sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
-        expect(user.notifications.count).to eq(0)
+        welcome_thread_article = create(:article, title: "Welcome Thread - v0", published: true, tags: "welcome", user: mascot_account)
+        create(:comment, commentable: welcome_thread_article, commentable_type: "Article", user: user)
+
+        expect do
+          sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+        end.to not_change(user.notifications, :count)
       end
 
       it "does not send a duplicate notification" do
-        2.times do
-          sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
-        end
-
-        expect(receiving_user.notifications.count).to eq(1)
+        sidekiq_perform_enqueued_jobs { 2.times { described_class.call(user.id) } }
+        expect(user.notifications.count).to eq(1)
       end
+
+      it_behaves_like "unsubscribed from welcome notifications"
```

Comment:
what does this do? RSpec magic?

Reply:
It applies [this shared example](https://github.com/thepracticaldev/dev.to/pull/6852/files#diff-133a9d8432d8554d80ce66e17cd9a94cR5-R11) in the context of this block :)

Basically my thought here is that a lot of this file could benefit from shared examples since we always want to test that we don't send any kind of welcome notification to a user who is unsubscribed, that we don't send double notifications to a user for any welcome notification, etc etc.

Reply:
I'm curious about this, too @rhymes ! 🧙

Reply:
@juliannatetreault https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples